### PR TITLE
Add github action to publish to pypi

### PR DESCRIPTION
**Pull request by staylorx**
# Description

As discussed in discussion item #181:
This adds a GitHub action file to build and publish a source (bdist) and binary (wheel) distribution to PyPI.
On any release, the action will build and publish to the test PyPI repo.
On any tagged release the artifacts will be published to the production PyPI repo.
The /dist directory is added to the .gitignore file.
A prior PR (#187) was submitted, but the changes were too broad.

## What is the nature of your change?

- [ ] Bug fix (fixes an issue).
- [X] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [X] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Comment by staylorx** [comment]
My version of the .github/workflows/publish-to-pypi.yml file has an ending newline, which I added from comments from #187.

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: python -m pip install setuptools build wheel --user
+
+      - name: Build a binary wheel
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish any distribution 📦 to Test PyPI
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
```
Comment (gwaygenomics) [inline .github/workflows/publish-to-pypi.yml:30]: should this point to the pycytominer url? What is legacy?

**Review by gwaygenomics — APPROVED** [review]
Looks good

File: .github/workflows/publish-to-pypi.yml (hunk 1 above)
Comment (staylorx) [reply .github/workflows/publish-to-pypi.yml:30]: `repository_url` is the API endpoint for the test PyPI system. By default that URL points to production PyPI.

https://github.com/pypa/gh-action-pypi-publish

**Comment by gwaygenomics** [comment]
p.s. @niranjchandrasekaran - I am not sure why both of us aren't automatically assigned to review. We're both listed as [CODEOWNERS](https://github.com/cytomining/pycytominer/blob/master/.github/CODEOWNERS).

I wonder if we should disable this feature though - do you find it useful being tagged?

**Comment by niranjchandrasekaran** [comment]
> I am not sure why both of us aren't automatically assigned to review. We're both listed as [CODEOWNERS](https://github.com/cytomining/pycytominer/blob/master/.github/CODEOWNERS).

Not sure but it seems like it always assigns me.

> I wonder if we should disable this feature though - do you find it useful being tagged?

We could disable it as long as we are keeping track of PRs and determine who is going to review it.

**Pull request merged**